events/src/main/java/com/app/events/serviceimpl/SectorServiceImpl.java

```diff
@@ -54,27 +54,28 @@ public Sector create(Sector sector) throws Exception{
 
     @Override
     public Sector update(Sector sector) throws Exception{
-        Optional<Sector> sectorOpt = this.sectorRepository.findById(sector.getId());
-        if( !sectorOpt.isPresent())
-        {
-            throw new SectorDoesntExistException("Not found sector."); 
-        }
-        Sector sectorToUpdate = sectorOpt.get();
+        Sector sectorToUpdate = this.findOne(sector.getId());
         Optional<Hall> hallOpt = this.hallRepository.findById(sector.getHall().getId());
         if( !hallOpt.isPresent())
         {
-            throw new HallDoesntExist("Not found hall."); // custom exception here!
+            throw new HallDoesntExist("Not found hall.");
         }
-        sectorToUpdate.setHall(hallOpt.get());
-        sectorToUpdate.setName(sector.getName());
-        sectorToUpdate.setSectorColumns(sector.getSectorColumns());
-        sectorToUpdate.setSectorRows(sector.getSectorRows());
+        sectorToUpdate = this.prepareSectorFields(sectorToUpdate, sector, hallOpt);
```

Comment:
mozda bi bilo bolje da Sector klasa ima metodu update ili tako nesto?
npr
public void update(Sector s) {
this.name = s.getName();
....
}
I zasto sektor moze da menja Halu?

Reply:
namestio sam metodu u servisu za setovanje, msm da je okej i tu da bude.